packages/ra-core/src/auth/useCanAccessResources.ts

```diff
@@ -75,10 +71,22 @@ export const useCanAccessResources = <
                     });
 
                     return { canAccess, resource };
-                },
-            };
-        }),
-        combine: combineSourceAccessResults<ErrorType>,
+                })
```

Comment:
suggestion: *reduce* that array of objects to make it an object with only the allowed resources. This will make the lookup below much simpler, and more effective (n finds : O(n2) => n object lookups : O(n))

Reply:
Done